pkg/controller/monitor/monitor_controller.go

```diff
@@ -225,6 +231,30 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 			return reconcile.Result{}, err
 		}
 	}
+	preDefaultPatchFrom := client.MergeFrom(instance.DeepCopy())
+	fillDefaults(instance)
+	// Patch the monitor resource with defaults added.
+	if err = r.client.Patch(ctx, instance, preDefaultPatchFrom); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Failed to write defaults", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	var externalPrometheus bool
+	if instance.Spec.ExternalPrometheus != nil {
+		if err = r.client.Get(ctx, client.ObjectKey{Name: instance.Spec.ExternalPrometheus.Namespace}, &corev1.Namespace{}); err != nil {
+			if !errors.IsNotFound(err) {
+				r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get external prometheus namespace", err, reqLogger)
+				return reconcile.Result{}, err
+			}
+			// We set ExternalPrometheus to nil, and proceed to render the other configuration.
+			// When the namespace is created, we will reconcile the ExternalPrometheus.
```

Comment:
Is this extra complexity needed? Why not just block progression if an ExternalPrometheus namespace was provided, but the namespace doesn't exist?

At a minimum, that feels like something we should be emitting in TigeraStatus to alert the user of invalid configuration (which it is).